### PR TITLE
handle unload

### LINKED ISSUE #6995: amp-viewer-integration extension handle unload
handle amp document unload

https://github.com/ampproject/amphtml/issues/4997

### PR DESCRIPTION
**Pull request by chenshay**
close #6995

### CODE REVIEW

File: extensions/amp-viewer-integration/0.1/amp-viewer-integration.js [hunk 1]
```diff
@@ -77,7 +89,16 @@ export class AmpViewerIntegration {
...
-    return messaging.sendRequest('channelOpen', {}, true);
+    return messaging.sendRequest(RequestNames.CHANNEL_OPEN, {}, true);
+  }
+
+
+  /**
+   * Notifies the viewer when this document is unloaded.
+   * @return {Promise<*>|undefined}
+   */
+  handleUnload(messaging) {
```
Comment (dvoytenko) [inline extensions/amp-viewer-integration/0.1/amp-viewer-integration.js]: Should be `@private` with `_` suffix.

File: extensions/amp-viewer-integration/0.1/amp-viewer-integration.js [hunk 2]
```diff
@@ -77,7 +89,16 @@ export class AmpViewerIntegration {
...
    */
   openChannel(messaging) {
-    return messaging.sendRequest('channelOpen', {}, true);
+    return messaging.sendRequest(RequestNames.CHANNEL_OPEN, {}, true);
+  }
+
+
+  /**
+   * Notifies the viewer when this document is unloaded.
+   * @return {Promise<*>|undefined}
```
Comment (dvoytenko) [inline extensions/amp-viewer-integration/0.1/amp-viewer-integration.js]: `@param` missing.

File: extensions/amp-viewer-integration/0.1/amp-viewer-integration.js [hunk 3]
```diff
@@ -68,6 +77,9 @@ export class AmpViewerIntegration {
 
           viewer.setMessageDeliverer(messaging.sendRequest.bind(messaging),
             dev().assertString(this.unconfirmedViewerOrigin_));
+
+          listenOnce(
```
Comment (dvoytenko) [inline extensions/amp-viewer-integration/0.1/amp-viewer-integration.js]: Tests needed. Might be of help - we expose `window.eventListeners` for testing purposes in `realWin` and `fakeWin` fixtures. E.g. you should be able to do something like:
```
win.eventListeners.fire({type: 'unload'});
```

File: extensions/amp-viewer-integration/0.1/amp-viewer-integration.js (hunk 1 above)
Comment (chenshay) [inline extensions/amp-viewer-integration/0.1/amp-viewer-integration.js]: done

File: extensions/amp-viewer-integration/0.1/amp-viewer-integration.js (hunk 2 above)
Comment (chenshay) [inline extensions/amp-viewer-integration/0.1/amp-viewer-integration.js]: done

File: extensions/amp-viewer-integration/0.1/amp-viewer-integration.js (hunk 3 above)
Comment (chenshay) [inline extensions/amp-viewer-integration/0.1/amp-viewer-integration.js]: done

**Pull request merged**